src/main/resources/db/migration/V1_95_1__referral_details_table.sql

```diff
@@ -0,0 +1,25 @@
+CREATE TABLE referral_details(
+    id uuid not null,
+    superseded_by_id uuid unique,
+    created_at timestamp with time zone not null,
+    created_by text not null,
+    reason_for_change text,
+    referral_id uuid not null,
+    completion_deadline date,
+    further_information text,
+    maximum_enforceable_days int,
+
+    primary key(id),
+    foreign key (superseded_by_id) references referral_details(id)
+);
+create unique index idx_referral_detail_referral_id on referral_details (referral_id);
```

Comment:
`idx_referral_detail_referral_id` cannot be a unique index if we have `superseded_by_id` as we would have multiple referral_ids per "version"